src/libraries/System.Net.Http/src/System/Net/Http/Headers/HttpHeaders.cs

```diff
@@ -1165,7 +1165,7 @@ internal static void GetStoreValuesAsStringOrStringArray(HeaderDescriptor descri
                 else
                 {
                     Debug.Assert(length > 1, "The header should have been removed when it became empty");
-                    values = multiValue = new string[length];
+                    values = (multiValue = new string[length])!;
```

Comment:
Nullability changed a bit where previously user-defined conversions between Spans and arrays were involved and now the built-in implicit span conversions are used. The rules should be actually "better", for example this code was assigning `string[]` to `Span<string?>` without nullability warnings.